temporalio/testing/_activity.py

```diff
@@ -74,15 +74,30 @@ def __init__(self) -> None:
         self._cancelled = False
         self._worker_shutdown = False
         self._activities: Set[_Activity] = set()
+        self.cancellation_details: (
+            temporalio.activity._ActivityCancellationDetailsHolder
+        )
 
-    def cancel(self) -> None:
+    def cancel(
+        self,
+        cancellation_details: Optional[
+            temporalio.activity.ActivityCancellationDetails
+        ] = None,
+    ) -> None:
         """Cancel the activity.
 
+        Args:
+            cancellation_details: Optional details about the cancellation. When provided, these
+                will be accessible through temporalio.activity.cancellation_details()
+                in the activity after cancellation.
+
         This only has an effect on the first call.
         """
         if self._cancelled:
             return
         self._cancelled = True
+        if cancellation_details:
+            self.cancellation_details.set_details(cancellation_details)
```

Comment:
I think this needs to always be set, even if not provided. May want to change param to be non-option w/ a default of `temporalio.activity.ActivityCancellationDetails()`. Usually using a default param of a full expression is bad in Python because it isn't called for each call, but since the details are immutable, sharing the default for everyone is safe.

Reply:
I think the logic needs to be "you can guarantee `activity.cancellation_details()` is not `None` when the activity is being canceled", but this breaks that guarantee.